woocommerce/woocommerce
# Rating Filter: first iteration of Minimum Rating feature

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





This PR does the first implementation of the Minimum Rating featured, described in [this issue](https://github.com/woocommerce/woocommerce/issues/51789).

Relevant changes:

* Introduce the new `minRating` block attribute
* Render the `Minimum rating` in the block sidebar
* Store minimum rating value selected by the user
* Render the Rating Filters representation in the editor canvas

Disclaimer: Let's handle the block representation in the frontend in a follow-up task

Part of #51789
Closes # .



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

0. Ensure the experimental blocks are enabled:
    - If you use the WooCommerce Beta Tester plugin, you can enable the experimental blocks in Tools > WCA Test Helper > Features. Toggle the `experimental-blocks` option.
    - If you build this PR locally, build the branch with `pnpm --filter='@woocommerce/plugin-woocommerce' build` command to have experimental blocks available.
1. Ensure to add a few products to your store
2. Ensure to rating products
3. Add the **Product Filters (Experimental)** block to the Product Catalog template.
4. Confirm you see the `Minimum rating` panel in the block sidebar:
5. Confirm you change the minimum value
6. Confirm the block representation in the canvas changes accordingly
7. Save the post.
8. Hard refresh
9. Confirm the minimum rating option is persistent. The application shows the last selected value


https://github.com/user-attachments/assets/a785ca3d-8d0b-4e54-8bad-4574783c8de2




### Changelog entry







<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
